source/grid/tria.cc

```diff
@@ -12520,7 +12520,7 @@ Triangulation<dim, spacedim>::n_faces() const
   switch (dim)
     {
       case 1:
-        return 0;
+        return n_vertices();
       case 2:
         return n_lines();
```

Comment:
This is the implementation of the `n_faces()` function. Note how in 2d, you get `n_lines()`, which is documented as follows:
```
   * Return the total number of used lines, active or not.
```
On the other hand, `n_vertices()` is documented like this:
```
   * Return the total number of vertices.  Some of them may not be used, which
   * usually happens upon coarsening of a triangulation when some vertices are
   * discarded, but we do not want to renumber the remaining ones, leading to
   * holes in the numbers of used vertices.  You can get the number of used
   * vertices using @p n_used_vertices function.
```
So `n_vertices()` does not take the *used* status into account.

What you want to call here is `n_used_vertices()`, which is described as follows:
```
   * Return the number of vertices that are presently in use, i.e. belong to
   * at least one used element.
```

Reply:
Ahhh.  Now I see your point. Sorry!

👍